pczt/src/common.rs

```diff
@@ -0,0 +1,44 @@
+/// Global fields that are relevant to the transaction as a whole.
+#[derive(Clone)]
+pub(crate) struct Global {
+    //
+    // Transaction effecting data.
+    //
+    // These are required fields that are part of the final transaction, and are filled in
+    // by the Creator when initializing the PCZT.
+    //
+    pub(crate) tx_version: u32,
+    pub(crate) version_group_id: u32,
+    /// The consensus branch ID for the chain in which this transaction will be mined.
+    ///
+    /// Non-optional because this commits to the set of consensus rules that will apply to
+    /// the transaction; differences therein can affect every role.
+    pub(crate) consensus_branch_id: u32,
+    /// TODO: In PSBT this is `fallback_lock_time`; decide whether this should have the
+    /// same semantics.
+    pub(crate) lock_time: u32,
+    pub(crate) expiry_height: u32,
```

Comment:
It could be nice to type these as [`LockTime`](https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/transaction/lock_time.rs#L25) and `BlockHeight`.

> Should these be independently optional?

If this needs to be able to produce v1 or v2 transactions, then at least `expiry_height` should probably be optional.

Reply:
@arya2 I'm intentionally _not_ adding types to the core structure (at least at this stage), to keep its dependencies as minimal / non-existent as possible (because there are many roles that do not need to care about every type in the PCZT, e.g. an Orchard signer doesn't need to be able to parse Sapling-specific types). The Transaction Extractor role necessarily performs all conversions and checks, so that acts as a backstop.

Once we've figured out and agreed on the structure, then it may make sense to add certain newtypes that `pczt` can rely on (depending on who the downstream consumers of the crate will be).

Reply:
@arya2 I don't think it does need to be able to produce v1 or v2 transactions, or v3 for that matter. I only know of use cases for using PCZTs to produce new transactions, which would necessarily have to be at least v4.

Reply:
- `lock_time: u32` is now `fallback_lock_time: Option<u32>` and has the same semantics as in BIP 370.
- `expiry_height` remains a `u32` and is set by the Creator. (It could in theory be altered by an Updater prior to the IO Finalizer or Signers running, but we require it to always be set.)